content/articles/cancer-cell-classification-using-scikit-learn/index.md

```diff
@@ -0,0 +1,155 @@
+```
+There are 569 instances of tumor data in the dataset in question, each with a numerical value for one of the 30 variables in question. Data you have acquired indicates that the first tumor is malignant and has a mean radius of 1.79900000e+01 based on the information you know.
+
+#### 4. Creating sets of data to organize the information
+We need to evaluate our classifier's accuracy by running it on data that has never been seen before.This means we'll divide our data into two sets, the training and test sets, before we begin creating our model. Training and evaluation of the model will be done on the training set, and then it will be used to predict the unknown test set.
+
+Sklearn comes with a function called **train test split()**, which automatically divides the data into these two groups for us.
+
+```python 
+from sklearn.model_selection import train_test_split
+
+train, test, trainingLabels, testingLabels = train_test_split(ourfeatures, ourlabels, test_size = 0.33, random_state = 42)
+```
+It is possible to utilize the test size as a parameter to randomly divide the data. We've divided the original data into test data by a third of its original size. Other than that, the rest of the data is used for training. A different label for the train and test variables is used for each.
+
+Refer to the [documentation](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) for further information on how to use **train_test_split**() function.
+
+#### 5. Building the Model
+You can select from a wide variety of machine learning models to meet your needs. Each one has advantages and cons of its own. A popular method for binary classification, dubbed `Naive Bayes,` has been selected for use in this model. GaussianNB is the first module that needs to be imported and initialized using **GaussianNB()** function. then use the **fit()** method to train your model on the dataset's data.
+
+```python
+from sklearn.naive_bayes import GaussianNB
+
+gaussiannb = GaussianNB()
+ourModel = gaussiannb.fit(train, trainingLabels)
+```
+To generate predictions on our test set, we first need to finish training the model. The built-in **predict()** function returns a prediction value array for each data point in the test set. Our forecasts will be displayed using the **print()** method.
+
+```python
+ourPredictions = gaussiannb.predict(test)
+print(ourPredictions)
+```
+#### 6. Accuracy testing of the model trained
+If we want to make sure our model is accurate, we may compare the predictions it makes to the actual labels in the testing set. Use the sklearn module built-in **accuracy_score()** function for this task.
+
+```python
+from sklearn.metrics import accuracy_score
+
+print(accuracy_score(testingLabels, ourPredictions))
+```
+Output:
+
+```bash
+0.9414893617021277
+```
+If a tumor is malignant or benign, our studies show that this machine learning classifier based on the `Naive Bayes` algorithm is 94.15% accurate.

```

Comment:
So, is the tumor malignant or benign from the results obtained? This is not clearly stated.

Reply:
Already stated that it's malignant when I printed the features.